network/p2p/ping.go

```diff
@@ -0,0 +1,174 @@
+package p2p
+
```

Comment:
Loosely based on how libp2p implements it's `Ping` protocol - https://github.com/libp2p/go-libp2p/blob/master/p2p/protocol/ping/ping.go